src/densenet.jl

```diff
@@ -1,92 +1,188 @@
-struct Bottleneck
-    layer
+"""
+    dense_bottleneck(inplanes, growth_rate)
+
+Create a Densenet bottleneck layer
+([reference](https://arxiv.org/abs/1608.06993)).
+
+# Arguments
+- `inplanes`: number of input feature maps
+- `growth_rate`: number of output feature maps
+                 (and scaling factor for inner feature maps; see ref)
+"""
+function dense_bottleneck(inplanes, growth_rate)
+  inner_channels = 4 * growth_rate
+  m = Chain(conv_bn((1, 1), inplanes, inner_channels; usebias=false, rev=true)...,
+            conv_bn((3, 3), inner_channels, growth_rate; pad=1, usebias=false, rev=true)...)
+
+  SkipConnection(m, (mx, x) -> cat(x, mx; dims=3))
 end
 
-@functor Bottleneck
+"""
+    transition(inplanes, outplanes)
 
-Bottleneck(in_planes, growth_rate) = Bottleneck(Chain(BatchNorm(in_planes, relu),
-                                          Conv((1, 1), in_planes => 4growth_rate),
-                                          BatchNorm(4growth_rate, relu),
-                                          Conv((3, 3), 4growth_rate => growth_rate, pad = (1, 1))))
+Create a DenseNet transition sequence
+([reference](https://arxiv.org/abs/1608.06993)).
 
-(b::Bottleneck)(x) = cat(b.layer(x), x, dims = 3)
+# Arguments
+- `inplanes`: number of input feature maps
+- `outplanes`: number of output feature maps
+"""
+transition(inplanes, outplanes) = (conv_bn((1, 1), inplanes, outplanes; usebias=false, rev=true)...,
```

Comment:
```suggestion
transition(inplanes, outplanes; bias = true, rev = true) = (conv_bn((1, 1), inplanes, outplanes; bias, rev)...,
```

Reply:
It doesn't really make sense for `rev` to be user configurable here. I want to save `bias` for a later PR, so we can come up with a generic way to specify the convolution layer parameters across all model types. But I think that's an enhancement better saved for the future to prevent blocking current work in other packages. It's not like the current Metalhead allows this configuration, so it isn't a regression.